modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AMQPMetaDataHandler.java

```diff
@@ -55,4 +66,38 @@
 
         return underlying;
     }
+
+
+    /**
+     * Construct metadata while creating a copy of MessageMetadata for AMQP; after an update of the compression
```

Comment:
Would it be better to mention where the constructMetadata(..) overloads be applicable i.e the function overload A is used when invoked from UI etc